tests/test_healthcheck.py

```diff
@@ -128,29 +130,40 @@ def test_check_fails_if_at_least_check_for_one_item_fails(self):
 
 
 class TestFilesExistHealthCheck(TestCase):
-    @patch('os.path.isfile')
-    def test_ok_if_all_files_exist(self, is_file_mock):
-        is_file_mock.return_value = True
-        check = FilesExistHealthCheck(('file1', 'file2'), check_id='checkid')
+    def test_ok_if_all_files_exist(self):
+        tmpfile1 = NamedTemporaryFile()
+        tmpfile2 = NamedTemporaryFile()
+        file1 = tmpfile1.name
+        file2 = tmpfile2.name
+        check = FilesExistHealthCheck((file1, file2), check_id='checkid')
         check.run()
         self.assertTrue(check.is_ok)
-        self.assertEqual(check.details, {'file1': 'exists',
-                                         'file2': 'exists'})
+        self.assertEqual(check.details, {file1: 'exists',
+                                         file2: 'exists'})
 
-    @patch('os.path.isfile')
-    def test_ok_if_at_least_one_file_doesnt_exist(self, is_file_mock):
-        is_file_mock.side_effect = lambda fn: False if fn == 'file2' else True
-        check = FilesExistHealthCheck(('file1', 'file2'), check_id='checkid')
+    def test_not_ok_if_at_least_one_file_doesnt_exist(self):
+        tmpfile1 = NamedTemporaryFile()
+        file1 = tmpfile1.name
+        check = FilesExistHealthCheck((file1, 'file2'), check_id='checkid')
         check.run()
         self.assertFalse(check.is_ok)
-        self.assertEqual(check.details, {'file1': 'exists',
+        self.assertEqual(check.details, {file1: 'exists',
                                          'file2': 'NO SUCH FILE'})
 
+    @patch('os.stat')
+    def test_ok_if_file_exists_with_wrong_permissions(self, stat_mock):
+        stat_mock.side_effect = OSError(errno.EACCES, 'Permission denied')
+        self.test_ok_if_all_files_exist()
```

Comment:
> For me "call another test, which check different thing, but still ok for us, and see what happens" logic is much harder to understand than "this check must fail".

You seem to have misunderstood something. The check must succeed. Calling the test must not throw an exception. It is checking the exact same thing: that the files exist. It isn't some kind of coincidence that the other test provides the same functionality.

> True/False and checking .ok is not important in this case.

It's extremely important. That's the only way to know if the check has done the right thing. You cannot write a test for this without it.

I find that this discussion is not a good use of time and would prefer not to continue unless someone else weighs in.